src/jlapi.c

```diff
@@ -88,7 +88,20 @@ JL_DLLEXPORT void jl_init_with_image(const char *julia_bindir,
     if (jl_is_initialized())
         return;
     libsupport_init();
-    jl_options.julia_bindir = julia_bindir;
+    if (julia_bindir) {
+        jl_options.julia_bindir = julia_bindir;
+    } else {
+        char *libbindir = NULL;
+#ifdef _OS_WINDOWS_
+        libbindir = strdup(jl_get_libdir());
+#else
+        (void)asprintf(&libbindir, "%s" PATHSEPSTRING ".." PATHSEPSTRING "%s", jl_get_libdir(), "bin");
+#endif
+        if (!libbindir) {
+            printf("jl_init unable to find libjulia!\n");
+            abort();
+        }
```

Comment:
Oops, don't forget to update `jl_options.julia_bindir`:
```suggestion
#ifdef _OS_WINDOWS_
        jl_options.julia_bindir = strdup(jl_get_libdir());
#else
        int written = asprintf(&jl_options.julia_bindir, "%s" PATHSEPSTRING ".." PATHSEPSTRING "%s", jl_get_libdir(), "bin");
        if (written < 0) 
            abort(); // unexpected: memory allocation failed
#endif
```

The error check for `asprintf` needs to be this way according to https://man7.org/linux/man-pages/man3/asprintf.3.html

Also `jl_get_libdir()` will panic / abort rather than return `NULL`, so no need to handle that error